src/containers/Home.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { fetchPetitions, clearPetitions } from 'actions/PetitionsActions';
+import Home from 'components/Home';
+import getPetitions from 'selectors/petitions';
+import getPetitionsHomeQuery from 'selectors/petitionsHomeQuery';
+import isClientSideRouting from 'helpers/isClientSideRouting';
+
+const HOME_QUERY = getPetitionsHomeQuery({ limit: 3 });
+
+const HomeContainer = withRouter(React.createClass({
+
+  componentWillMount () {
+    // If there are no petitions, or if the user arrived on the page by clicking
+    // a client-side router link, then we fetch petitions client-side.
+    if (!this.props.petitions.length || isClientSideRouting(this.props.location)) {
```

Comment:
Ok, took a little time, but there's a scalable way to add groupings of petitions now. A helper called `helpers/getPetitionsGroupsList` has an array of groups, like `latest`, `trending`, or whatever else comes. This array determines the added properties to the home page `petitions` state object. Will probably need some tweaking in the future, but this serves what we need rn.